### DISCUSSION TITLE
Media (audio/video) player support

### DISCUSSION THREAD

**Discussion by pwfulghum** [dotnet/maui#6026] [Q&A]
I want to jump on the MAUI bandwagon, but my searching shows that there is no direct support for media (audio / video) playback.

Am I missing something??

Is there a nugget package that will work across iOS / PC / Android / Mac and be compatible with MAUI?

**Comment by pwfulghum** [comment]
2 weeks ....   Can somebody help me out, I really want to jump on .net MAUI, but media is fundamental for my app.

**Comment by xtuzy** [comment]
At xamarin, use vlc

**Reply by pwfulghum** [reply]
@Xtuzy, trying to use MAUI not Xamarin...  I see lots of example of VLC support for xamarin, but not for MAUI  (nor WinUI).

**Comment by pictos** [comment]
Hey @pwfulghum you can use VLC lib. Here's a project that you can use [as reference](https://github.com/drasticactions/DrasticMedia) made by @drasticactions

**Reply by KarlSearl** [reply]
xamarin forms had media element. 

this was moved the the community toolkit for 5.0,  I also need a media player in a production app, however I am not sure its been implemented yet. #

https://docs.microsoft.com/en-us/xamarin/community-toolkit/views/mediaelement

**Reply by pictos** [reply]
@KarlSearl you're right, he can use the `MediaElement` from `XCT.MauiCompat`, but that could not work very well since it uses the Compatibility layer.
And no, it will be ported after .NET MAUI GA

**Reply by pwfulghum** [reply]
@pictos, First off thank you....  I looked through that repo and I see where the VLCMediaService builds a VLCPlayer and plays, but I can't figure out how it actually gets DISPLAYED on a control/frame/image/etc.   There is no XAML in this repo.

There is no mediaelement in WinUI 3.0  (on the roadmap for "later").   VLC doesn't support WinUI (yet).
And I don't see a mediaelement in the toolkit.

I don't understand why these new technologies (Maui / WinUI) don't support video playback directly.

Any help would be appreciated.

**Reply by Mephisztoe** [reply]
The MediaElement control from within the XTC mauicompat will only work with iOS atm. On Android, initialization of the XAML control raises a Java Exception. However, there is still the Xamarin Plugin ExoPlayer for Android that still works in MAUI as an alternative. Both not very conveniant but if you implement them using a clean service layer with an interface, you could still get going and replace them later when there is 1st class mediaplayer support in .NET MAUI (I guess it will be still the MAUI Community Toolkit first).
